src/benchmarklib/synthetic_table_generator.cpp

```diff
@@ -0,0 +1,200 @@
+      node_id = (node_id + 1) % Topology::get().nodes().size();
+
+      resolve_data_type(column_data_types[column_index], [&](const auto column_data_type) {
+        using ColumnDataType = typename decltype(column_data_type)::type;
+
+        // get typed allocators
+        auto allocator_value_segment =
+            get_allocator_for_type<ValueSegment<ColumnDataType>>(node_id, numa_distribute_chunks);
+        auto allocator = get_allocator_for_type<ColumnDataType>(node_id, numa_distribute_chunks);
+
+        std::vector<int> values;
+        values.reserve(chunk_size);
+        const auto& column_data_distribution = column_data_distributions[column_index];
+
+        // generate distribution from column configuration
+        switch (column_data_distribution.distribution_type) {
+          case DataDistributionType::Uniform: {
+            auto uniform_dist = boost::math::uniform_distribution<double>{column_data_distribution.min_value,
+                                                                          column_data_distribution.max_value};
+            generate_value_by_distribution_type = [uniform_dist, &probability_dist, &pseudorandom_engine]() {
+              const auto probability = probability_dist(pseudorandom_engine);
+              return static_cast<int>(std::floor(boost::math::quantile(uniform_dist, probability)));
+            };
+            break;
+          }
+          case DataDistributionType::NormalSkewed: {
+            auto skew_dist = boost::math::skew_normal_distribution<double>{column_data_distribution.skew_location,
+                                                                           column_data_distribution.skew_scale,
+                                                                           column_data_distribution.skew_shape};
+            generate_value_by_distribution_type = [skew_dist, &probability_dist, &pseudorandom_engine]() {
+              const auto probability = probability_dist(pseudorandom_engine);
+              return static_cast<int>(std::round(boost::math::quantile(skew_dist, probability) * 10));
+            };
+            break;
+          }
+          case DataDistributionType::Pareto: {
+            auto pareto_dist = boost::math::pareto_distribution<double>{column_data_distribution.pareto_scale,
+                                                                        column_data_distribution.pareto_shape};
+            generate_value_by_distribution_type = [pareto_dist, &probability_dist, &pseudorandom_engine]() {
+              const auto probability = probability_dist(pseudorandom_engine);
+              return static_cast<int>(std::floor(boost::math::quantile(pareto_dist, probability)));
+            };
+            break;
+          }
+        }
+
+        // Generate values according to distribution. We first add min and max values to avoid hard-to-control
+        // pruning via dictionaries. In the main loop, we then run (num_rows/chunk_size)-2 times.
```

Comment:
-2?

Reply:
Better? 
```
/**
        * Generate values according to distribution. We first add the given min and max values of that column to avoid
        * early exists via dictionary pruning (no matter which values are later searched, the local segment
        * dictionaries cannot prune them early). In the main loop, we then execute the loop two times less.
        **/
```

Reply:
we **thus**?